ocaml-lsp-server/test/e2e/__tests__/uri.test.ts

```diff
@@ -0,0 +1,48 @@
+import outdent from "outdent";
+import * as rpc from "vscode-jsonrpc/node";
+import * as LanguageServer from "../src/LanguageServer";
+import * as Protocol from "vscode-languageserver-protocol";
+import * as Types from "vscode-languageserver-types";
+import { URI } from "vscode-uri";
+
+describe("uri", () => {
+  let languageServer: rpc.MessageConnection;
+  const uri = URI.file("src/néw/Mödel + Other Thîngß/test.ml").toString();
+
+  function openDocument(source: string) {
+    languageServer.sendNotification(
+      Protocol.DidOpenTextDocumentNotification.type,
+      {
+        textDocument: Types.TextDocumentItem.create(uri, "ocaml", 0, source),
+      },
+    );
+  }
+
+  beforeEach(async () => {
+    languageServer = await LanguageServer.startAndInitialize();
+  });
+
+  afterEach(async () => {
+    await LanguageServer.exit(languageServer);
+  });
+
+  it("handles uri with special characters", async () => {
+    let receivedDiganostics: Promise<Protocol.PublishDiagnosticsParams> =
+      new Promise((resolve, _reject) =>
+        languageServer.onNotification(
+          Protocol.PublishDiagnosticsNotification.type,
+          (params) => {
+            resolve(params);
+          },
+        ),
+      );
+    openDocument(outdent`
+      let () =
+        let x = 123 in
+        ()
+    `);
+    const params = await receivedDiganostics;
+
+    expect(params.uri).toBe(uri);
```

Comment:
It fails with:

```js
Expected: "file:///src/n%C3%A9w/M%C3%B6del%20%2B%20Other%20Th%C3%AEng%C3%9F/test.ml"
Received: "file:///src/n%C3%A9w/M%C3%B6del %2B Other Th%C3%AEng%C3%9F/test.ml"
```

https://github.com/ocaml/ocaml-lsp/pull/739 should fix it but I'm not sure if we really need an e2e test for that since we already have OCaml tests for the URI parsing?